system/Images/Handlers/ImageMagickHandler.php

```diff
@@ -232,9 +232,10 @@ public function getVersion(): string
 	 * @param string  $action
 	 * @param integer $quality
 	 *
-	 * @return ImageMagickHandler|boolean
+	 * @return array  Lines of output from shell command
+	 * @throws \Exception
 	 */
-	protected function process(string $action, int $quality = 100)
+	protected function process(string $action, int $quality = 100): array
```

Comment:
Are you sure this only ever returns an array? I think there might be instances where `exec` returns something other than an array at times.

Reply:
It isn't the actual return, but an array that you pass in to receive the lines of output. It's actually passed by reference so will definitely always be an array, and that's what we're returning (though there could be arguments for `process()` returning something else).